var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,531 @@
+            description='Enable Fortran 90 examples')
+    variant('examples-cuda', default=False,
+            description='Enable CUDA examples')
+    variant('examples-raja', default=False,
+            description='Enable RAJA examples')
+    variant('examples-install', default=True,
+            description='Install examples')
+
+    # Generic (std-c) math libraries (UNIX only)
+    variant('generic-math', default=True,
+            description='Use generic (std-c) math libraries on unix systems')
+
+    # xSDK options
+    variant('xsdk-defaults', default=False,
+            description='Use default xSDK configuration')
+
+    # ==========================================================================
+    # Conflicts
+    # ==========================================================================
+
+    # Options added after v2.6.2
+    conflicts('+hypre', when='@:2.6.2')
+    conflicts('+petsc', when='@:2.6.2')
+
+    # Options added after v2.7.0
+    conflicts('+cuda',          when='@:2.7.0')
+    conflicts('+raja',          when='@:2.7.0')
+    conflicts('+blas',          when='@:2.7.0')
+    conflicts('+indextype',     when='@:2.7.0')
+    conflicts('+xsdk-defaults', when='@:2.7.0')
+    conflicts('+examples-cuda', when='@:2.7.0')
+    conflicts('+examples-raja', when='@:2.7.0')
+
+    # External libraries incompatible with 64-bit indices
+    conflicts('+lapack', when='@3.0.0-beta-2: indextype=int64_t')
+    conflicts('+hypre',  when='+hypre@:2.6.1a indextype=int64_t')
+
+    # External libraries incompatible with single precision
+    conflicts('+klu',   when='precision=single')
+    conflicts('+hypre', when='+hypre@:2.12.0 precision=single')
+
+    # External libraries incompatible with extended (quad) precision
+    conflicts('+lapack',     when='precision=extended')
+    conflicts('+superlu-mt', when='precision=extended')
+    conflicts('+klu',        when='precision=extended')
+    conflicts('+hypre',      when='+hypre@:2.12.0 precision=extended')
+
+    # ==========================================================================
+    # Dependencies
+    # ==========================================================================
+
+    # Build dependencies
+    depends_on('cmake', type='build')
+
+    # MPI related dependencies
+    depends_on('mpi', when='+mpi')
+    depends_on('mpi', when='+hypre')
+    depends_on('mpi', when='+petsc')
+
+    # Other parallelism dependencies
+    depends_on('cuda', when='+cuda')
+    depends_on('raja', when='+raja')
+
+    # External libraries
+    depends_on('hypre',              when='+hypre')
+    depends_on('petsc',              when='+petsc')
+    depends_on('blas',               when='+blas')
     depends_on('lapack',             when='+lapack')
     depends_on('suite-sparse',       when='+klu')
-    depends_on('superlu-mt+openmp',  when='+superlu+openmp')
-    depends_on('superlu-mt+pthread', when='+superlu+pthread')
+    depends_on('superlu-mt+openmp',  when='+superlu-mt+openmp')
```

Comment:
you probably want to add a conflict that `+superlu-mt` needs either `openmp` or `pthread`, and if both are disabled you can't have `sundials+supleru-mt`.

Reply:
These `depend` lines are not quite right since it is possible to use `superlu-mt` from Sundials without enabling `openmp` or `pthreads` (i.e. Sundials is run in serial but the linear solves with `superlu_mt` are threaded). 

So what I need is to know is if `superlu_mt` was build with `openmp` or `pthread` and set our CMake variable accordingly. Is this the right way to do that
```
if spec.satisfies('^superlu-mt+openmp'):
```

Reply:
Yes, I think it’s the right syntax

Reply:
The above if works as expected for checking how `superlu_mt` was configured.